pink/barriers/barrier.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Stéphane Caron
+
+"""All barriers derive from the :class:`Barrier` base class.
+
+The formalism used in this implementation is written down in
+examples/barriers/NOTES.md
```

Comment:
I think it will be better if we publish these notes somewhere and then provide the link, as [you did here](https://stephane-caron.github.io/pink/tasks.html)